### ISSUE TITLE
build error on debian sid

### ISSUE THREAD

**Issue by dromer** [rgbdemo/rgbdemo#27]
I've had this error before when building openni. https://groups.google.com/forum/?fromgroups=#!searchin/openni-dev/radu$20glut/openni-dev/qv7p-7jy5gQ/DUaa8FAKbPkJ

I couldn't figure out which Makefiles needed editting, so dumping it here :)

Linking CXX executable ../../../../../../bin/Sample-Players
cd /home/dreamer/Sources/rgbdemo/build/nestk/deps/openni/Nite/Samples/Players && /usr/bin/cmake -E cmake_link_script CMakeFiles/Sample-Players.dir/link.txt --verbose=1
/usr/bin/c++   -O2 -g    CMakeFiles/Sample-Players.dir/signal_catch.cpp.o CMakeFiles/Sample-Players.dir/main.cpp.o CMakeFiles/Sample-Players.dir/SceneDrawer.cpp.o CMakeFiles/Sample-Players.dir/kbhit.cpp.o  -o ../../../../../../bin/Sample-Players -rdynamic -L/home/dreamer/Sources/rgbdemo/nestk/deps/openni/Nite/Lib/Linux32 ../../../../../../lib/libOpenNI.so -lglut -lusb-1.0 -ldl ../../../../../../lib/libTinyXml.a -Wl,-rpath,/home/dreamer/Sources/rgbdemo/nestk/deps/openni/Nite/Lib/Linux32:/home/dreamer/Sources/rgbdemo/build/lib 
/usr/bin/ld: CMakeFiles/Sample-Players.dir/SceneDrawer.cpp.o: undefined reference to symbol 'glEnd'
/usr/bin/ld: note: 'glEnd' is defined in DSO /usr/lib/libGL.so.1 so try adding it to the linker command line
/usr/lib/libGL.so.1: could not read symbols: Invalid operation
collect2: error: ld returned 1 exit status

**Comment by nburrus** [comment]
Seems that your OpenGL install is kind of broken, did you mess up with it
somehow?

**Comment by dromer** [comment]
Nope. Supposedly there needs to be the added lib 'GL' in the linker.
Check Radu's post in the aforementioned OpenNI mailinglist-topic.

**Comment by nburrus** [comment]
I see. On which platform are you? This should be only required on Mac,
according to

if(DEFINED OPENNI_PLATFORM_APPLE)
        find_library(OPENGL_LIB OpenGL)
endif(DEFINED OPENNI_PLATFORM_APPLE)

in nestk/deps/Nite/Samples/Players/CMakeLists.txt

you can try to disable the IF if you are not on Mac to see if it helps.
Otherwise just disable temporarily OpenNI samples using
the OPENNI_BUILD_SAMPLES cmake variable...

**Comment by dromer** [comment]
Debian sid (gnu/linux), as said in the title :)

I don't have a nest/deps/Nite/ in my 'clone --recursive' of the repo.

I'll try the second option then.

**Comment by nburrus** [comment]
nestk/deps/openni/Nite, sorry. It missed the openni :)

**Comment by dromer** [comment]
Hmm, that didn't work. Also I can't find OPENNI_BUILD_SAMPLES in ccmake.

**Comment by nburrus** [comment]
Can you try cmake -DOPENNI_BUILD_SAMPLES=OFF . ?

**Comment by dromer** [comment]
Same ..

**Comment by ntx-80** [comment]
Just a quickie: Can you tell us what "file -L /usr/lib/libGL.so.1" says?

I'm suspecting that you're mixing 32 & 64 bit libraries in your link line.

**Comment by dromer** [comment]
$ file -L /usr/lib/libGL.so.1
/usr/lib/libGL.so.1: ELF 32-bit LSB shared object, Intel 80386, version 1 (SYSV), dynamically linked, stripped

Running on 32-bit (3.4.0-10.dmz.1-liquorix-686). As far as I know I don't have any 64-bit libraries installed.

Like I said before I had this same problem with OpenNI, which was resolved by a simple patch to some makefiles.
